kiwi/xml_description.py

```diff
@@ -52,9 +54,9 @@ class XMLDescription(object):
         path to base XML description file
 
     """
-    def __init__(self, description, derived_from=None):
+    def __init__(self, description=None, derived_from=None, content=None):
         self.description_xslt_processed = NamedTemporaryFile()
-        self.description = description
+        self.description = content and BytesIO(bytes(content)) or description
```

Comment:
I only see a problem with the use of self.description. The scope of it is now two fold, it could be either the path to a file name or raw content which is not a problem for etree.parse but at other places e.g if an exception occurs we use self.description in the exception message and it would make only sense if it would be in the scope of a filename. In addition the member variable parse.description_dir should be None if raw content will be used and would be set to the content which might be confusing

How about dealing with self.description and self.content instead of merging it at the object construction level
e.g

```python
self.description = description
self.content = content

...

etree.parse(self.content and BytesIO(bytes(self.content)) or self.description)

...handle output of KiwiDescriptionInvalid exception accordingly

```

I think that would put the logic at the place where it is needed and follows more the explicit is better than implicit python advise

Thoughts ?